### DISCUSSION TITLE
Provisioning Database server & Redis

### DISCUSSION THREAD

**Discussion by jasperf** [deployphp/deployer#3174] [Help needed]
Is there an example anywhere of a `provision.yml` which deploy a website with database, Node, PHP and Redis? What about two servers: database server and web server? 

When I check https://deployer.org/docs/7.x/recipe/provision I do see listed tasks, but no real example file as of yet.

```
provision
```

This task is group task which contains next tasks:

```yml
provision:check
provision:configure
provision:update
provision:upgrade
provision:install
provision:ssh
provision:firewall
provision:deployer
provision:server
provision:php
provision:databases
provision:composer
provision:npm
provision:website
provision:verify
```

1. How to create a database only server?
2. How to add Redis to server?
3. How to add daemons for queues?
4. How to run `provision.yml` and not the `deploy.yml`?

Database server  stub:

```yml
provision:check
provision:configure
provision:update
provision:upgrade
provision:install
provision:ssh
provision:firewall
provision:deployer
provision:server
//  provision:php probably not needed
provision:databases
provision:verify
```

Redis is installed by default at https://github.com/deployphp/deployer/blob/68a983a8ca52e6ffd44ad08f139d3dc94fb570da/recipe/provision.php#L131  under `install`  it seems. Creating a separate Redis server would be a different story I guess. But it is added with all the other packages.

**Comment by antonmedv** [comment]
Provision recipe create a server as a template. If you need different setup - create your own recipe.

**Reply by jasperf** [reply]
@antonmedv Thanks for the reply. So I need to create a recipe for custom server like the recipe for Laravel? Thought I could set up commands in a new `provision.yml`. Will look into that.
